### PR TITLE
enh(poller) Handle vault in broker input/output broker config for wizard poller

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Handle vault storage of db passwords when creating broker configurations for a new poller/remoteServer

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] master

### CODE REVIEW

File: centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php [hunk 1]
```diff
@@ -277,4 +276,72 @@ protected function isRemote(): bool
...
+        /** @var ReadVaultConfigurationRepositoryInterface $readVaultConfigurationRepository */
+        $readVaultConfigurationRepository = $kernel->getContainer()->get(
+            ReadVaultConfigurationRepositoryInterface::class
+        );
+        /** @var FeatureFlags $featureFlagManager */
+        $featureFlagManager = $kernel->getContainer()->get(FeatureFlags::class);
+
+        /** @var WriteVaultRepositoryInterface $writeVaultRepository */
+        $writeVaultRepository = $kernel->getContainer()->get(WriteVaultRepositoryInterface::class);
+        $writeVaultRepository->setCustomPath(AbstractVaultRepository::BROKER_VAULT_PATH);
```
Comment (Drago37) [inline centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php:299]: You can't pass these classes by a dependency injection ?

File: centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php [hunk 2]
```diff
@@ -277,4 +276,72 @@ protected function isRemote(): bool
...
+                continue;
+            }
+
+            $paths = $writeVaultRepository->upsert(
+                $this->uuid,
+                ["{$inputOutputName}_db_password" => $credentialValue]
+            );
+
+            if (end($paths) !== false) {
+                $this->uuid = $this->getUuidFromPath(end($paths));
```
Comment (Drago37) [inline centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php:339]: You call end() method two times, you can create a variable to store the return of end()  to optimize the code

File: centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php [hunk 3]
```diff
@@ -277,4 +276,72 @@ protected function isRemote(): bool
...
+            return $brokerInfos;
+        }
+
+        foreach ($brokerInfos as $key => $inputOutput) {
+            $inputOutputName = null;
+            $credentialKey = null;
+            $credentialValue = null;
+            foreach ($inputOutput as $index => $row) {
+                if ($row['config_key'] === 'name') {
+                    $inputOutputName = $row['config_value'];
```
Comment (Drago37) [inline centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php:317]: No isset() on $row['config_value'] ? Are you sure this array contains this key at this location ?

Comment (Drago37) [reply centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php:317]: thanks ;)

File: centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php (hunk 2 above)
Comment (Drago37) [reply centreon/src/CentreonRemote/Domain/Service/ConfigurationWizard/ServerConnectionConfigurationService.php]: ok

**Review by Drago37 — APPROVED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by Drago37 — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**